web/src/main/java/cz/muni/fi/pa165/team/TeamController.java

```diff
@@ -22,25 +30,45 @@
 {
 
     private final TeamFacade teamFacade;
+    private final TeamPlayerFacade teamPlayerFacade;
     private final TeamRepository teamRepository;
+    private final TeamMatchRepository teamMatchRepository;
+    private final TeamPlayerRepository teamPlayerRepository;
 
     @Autowired
-    public TeamController(TeamFacade teamFacade, TeamRepository teamRepository)
+    public TeamController(TeamFacade teamFacade,
+                          TeamPlayerFacade teamPlayerFacade,
+                          TeamRepository teamRepository,
+                          TeamMatchRepository teamMatchRepository,
+                          TeamPlayerRepository teamPlayerRepository
+    )
     {
         this.teamFacade = teamFacade;
+        this.teamPlayerFacade = teamPlayerFacade;
         this.teamRepository = teamRepository;
+        this.teamMatchRepository = teamMatchRepository;
+        this.teamPlayerRepository = teamPlayerRepository;
+    }
+
+    @InitBinder("teamDetailsRequest")
+    private void initBinder(WebDataBinder binder)
+    {
+        binder.setValidator(new TeamCreationValidator());
     }
 
     @RequestMapping(value = "/teams", method = RequestMethod.GET)
-    public ModelAndView viewList() throws Exception
+    public ModelAndView viewTeamsList() throws Exception
     {
+        Collection<TeamStatistics> teamsStatistics = new ArrayList<>();
+        Collection<Team> teamsList = teamRepository.findAll();
+        teamsStatistics.addAll(teamsList.stream().map(team -> teamFacade.getTeamStatistics(team.getId())).collect(Collectors.toList()));
```

Comment:
tohle proiteruje postupně snad úplně celou databázi... to by chtělo nějak lépe vymyslet... ale nevim jestli to teď už má smysl předělávat :-/